### PR TITLE
Fetch tags from publishing-api instead of content-api

### PR DESCRIPTION
**Pull request by tijmenb**
When editing an edition, editors can select a "specialist sector" (which we call topics now). They are currently loaded from the content-api, which is deprecated.

screen shot 2016-09-02 at 16 19 23

This commit makes the form use the publishing-api to fetch all topics, via the linkables endpoint.

I'm reasonably unhappy with the `LinkableTopics` class as it does a lot of things and has a bunch of magic-looking strings in there. I've tried to mitigate that with some commenting. In the short-to-medium term our team will [revisit this class](https://trello.com/c/4voUbx15) (and similar in mainstream publisher and content-tagger) to extract some common functionality for the population of select-boxes.

https://trello.com/c/pkERaVpJ

Follow up will be to remove whitehall's dependency in the development repo, and remove the `/tags.json` endpoint in content-api.

### CODE REVIEW

**Comment by boffbowsh** [comment]
Possibly beyond the scope of this PR, but would it be possible to store the content ids for topics against editions, instead of the topic slug? That way we don't need to look up the content id by base path in publishing api every time we do a publish or sync check

**Comment by tijmenb** [comment]
Yeah, I agree, that'll simplify a lot of things. I've made a follow up ticket for this here:https://trello.com/c/QYgPe6kn

File: features/support/specialist_sector_helper.rb [hunk 1]
```diff
@@ -11,33 +10,38 @@
...
-    artefact_slug = RegisterableEdition.new(document).slug
-    tag_slugs = document.specialist_sectors.map(&:tag)
-
-    # These methods are located in gds_api_adapters
-    stubbed_artefact = artefact_for_slug_with_a_child_tags('specialist_sector', artefact_slug, tag_slugs)
-    content_api_has_an_artefact(artefact_slug, stubbed_artefact)
+    publishing_api_has_linkables(
+      [
+        {
+          'base_path' => 'oil-and-gas/wells',
```
Comment (boffbowsh) [inline features/support/specialist_sector_helper.rb]: These will come back with absolute base_paths. Here's part of the response from publishing api:

```
{
  "title":"Oil and gas finance and taxation",
  "content_id":"037e8dfb-b6c3-4e9f-bdfb-6b3027211d99",
  "publication_state":"published",
  "base_path":"/topic/oil-and-gas/finance-and-taxation",
  "internal_name":"Oil and gas / Oil and gas finance and taxation"
}
```

Comment (tijmenb) [inline features/support/specialist_sector_helper.rb]: Good catch - updated.

**Comment by boffbowsh** [comment]
👍

**Pull request merged**